src/app/audiologist-navigation/audiologist-navigation.component.ts

```diff
@@ -47,7 +48,18 @@ export class AudiologistNavigationComponent {
 
   public submitSurvey() {
     if (this.summaryComponent != null) {
-      this.summaryComponent.submitSurvey();
+      this.summaryComponent.submitSurvey().subscribe(
+        _ => {
+          alert('Survey was Sucessfully Submitted');
+          this.logout();
+        },
+        catchError(
+          (error, _) => {
+            alert('Survey Failed to Submit');
+            throw error;
```

Comment:
Have you tested this case? Where is this error caught? Does the whole application explode?

Reply:
The error is caught by Angular's default error handler which displays the message to Chrome's console so the application remains stable. However, you are right, its NOT pretty. I'll remove this catchError here entirely. (This should be handled by an error service: _See [this branch](https://github.com/movshov/VA-Audiology-App/tree/CLIENT/ErrorHandler) to see what I hope to pull in after the demo.)_